modules/navier_stokes/src/fvkernels/INSFVMomentumAdvection.C

```diff
@@ -81,7 +81,8 @@ INSFVMomentumAdvection::computeResidualsAndAData(const FaceInfo & fi)
     const auto eps_face = epsilon()(ssf, state);
     const auto u_face = _var(ssf, state);
     const Real d_u_face_d_dof = u_face.derivatives()[dof_number];
-    const auto coeff = _normal * v_face * rho_face / eps_face;
+    const auto coeff = _linearize ? _normal * raw_value(v_face) * rho_face / eps_face
+                                  : _normal * v_face * rho_face / eps_face;
```

Comment:
should the linearization operation / information live in the RC UO entirely? in getVelocity

Reply:
Also is this a true linearization? I would have thought a linearization would use a previous iteration value, eg something where the Jacobian truly is linear; here the Jacobian is actually nonlinear in truth, it just won’t be in our representation of it. Seems like the worst of both worlds

Reply:
Ahhh yes, the value itself can change based on what we plug in the `computeResiudal/Jacobian` functions. Great point. At the same time, if the previous iterate is plugged in, then we should be good?  I suppose this can break down if we are doing things like pjfnk and line search where these are evaluated with perturbed vectors. I think I'll go with @GiudGiud 's suggestion, the users might not have to know this detail.